src/txmempool.cpp

```diff
@@ -1231,6 +1233,79 @@ size_t CTxMemPool::DynamicMemoryUsage() const {
     return total;
 }
 
+void CTxMemPool::UpdateMetrics() const {
+    LOCK(cs);
+
+    static const int64_t WEIGHT_RATIO_20_PCT = WEIGHT_RATIO_SCALE / 5;
+
+    // Track the sum of unpaid actions within each transaction (as they are subject to the
+    // unpaid action limit). Transactions that have weight >= 1 have no unpaid actions by
+    // definition.
+    size_t unpaidActionsWithWeightLt20pct = 0;
+    size_t unpaidActionsWithWeightLt40pct = 0;
+    size_t unpaidActionsWithWeightLt60pct = 0;
+    size_t unpaidActionsWithWeightLt80pct = 0;
+    size_t unpaidActionsWithWeightLt1 = 0;
+
+    // Track the total number of paid actions across all transactions in the mempool. This
+    // added to the bucketed unpaid actions above is equal to the total number of logical
+    // actions in the mempool.
```

Comment:
Nit that I noticed while preparing the `zebrad` PR:
```suggestion
    // added to the bucketed unpaid actions above is equal to the total number of
    // conventional actions in the mempool.
```

I'll fix this if/when we need to do another rebase, or once there is CI downtime (as this fix will require another round of CI).